contracts/modules/RelayerModule.sol

```diff
@@ -0,0 +1,372 @@
+            (isGuardian, guardians) = GuardianUtils.isGuardian(guardians, signer);
+            if (!isGuardian) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+    * @dev Refunds the gas used to the Relayer.
+    * @param _wallet The target wallet.
+    * @param _startGas The gas provided at the start of the execution.
+    * @param _gasPrice The gas price for the refund.
+    * @param _gasLimit The gas limit for the refund.
+    * @param _refundToken The token to use for the gas refund.
+    * @param _refundAddress The address refunded to prevents front-running.
+    */
+    function refund(
+        address _wallet,
+        uint _startGas,
+        uint _gasPrice,
+        uint _gasLimit,
+        address _refundToken,
+        address _refundAddress,
+        uint256 requiredSignatures,
+        OwnerSignature _ownerSignatureRequirement
+    )
+        internal
+    {
+        // only refund when approved by owner and positive gas price
+        if (_gasPrice == 0 || _ownerSignatureRequirement != OwnerSignature.Required) {
+            return;
+        }
+        address refundAddress = _refundAddress == address(0) ? msg.sender : _refundAddress;
+        uint256 gasConsumed = _startGas.sub(gasleft()).add(30000);
+        uint256 refundAmount;
+        // skip daily limit when approved by guardians (and signed by owner)
+        if (requiredSignatures > 1) {
+            refundAmount = Utils.min(gasConsumed, _gasLimit).mul(_gasPrice);
+        } else {
+            gasConsumed = gasConsumed.add(10000);
+            refundAmount = Utils.min(gasConsumed, _gasLimit).mul(_gasPrice);
+            uint256 ethAmount = LimitUtils.getEtherValue(tokenPriceStorage, refundAmount, _refundToken);
+            require(LimitUtils.checkAndUpdateDailySpent(limitStorage, _wallet, ethAmount), "RM: refund is above daily limt");
+        }
+        // refund in ETH or ERC20
+        if (_refundToken == LimitUtils.ETH_TOKEN) {
+            invokeWallet(_wallet, refundAddress, refundAmount, EMPTY_BYTES);
+        } else {
+            bytes memory methodData = abi.encodeWithSignature("transfer(address,uint256)", refundAddress, refundAmount);
+		    invokeWallet(_wallet, _refundToken, 0, methodData);
+        }
+        emit Refund(_wallet, refundAddress, _refundToken, refundAmount);
+    }
+
+   /**
+    * @dev Checks that the wallet address provided as the first parameter of the relayed data is the same
+    * as the wallet passed as the input of the execute() method.
+    @return false if the addresses are different.
+    */
+    function verifyData(address _wallet, bytes memory _data) private pure returns (bool) {
+        require(_data.length >= 36, "RM: Invalid dataWallet");
+        address dataWallet;
+        // solium-disable-next-line security/no-inline-assembly
+        assembly {
+            //_data = {length:32}{sig:4}{_wallet:32}{...}
+            dataWallet := mload(add(_data, 0x24))
```

Comment:
We could use the new slice syntax instead:
```
    function verifyData(address _wallet, bytes calldata _data) private pure returns (bool) {
        require(_data.length >= 36, "RM: Invalid dataWallet");
        (address dataWallet) = abi.decode(_data[4:], (address));
        return dataWallet == _wallet;
    }
```

Reply:
Nice and done!